### PR TITLE
fix(unions): Make unions render correctly when arrays are involved

### PR DESCRIPTION
**Pull request by ProbablePrime**
Here we update the union handling to be a bit better. A lot of this was a bit of a mess.The array handling was clobbering unions.

In addition unions in raml are without "()" [spec](https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md/#union-type)

The raml updates to make this render correctly are in: https://github.com/mixer/backend/pull/1464

### CODE REVIEW

File: src/rest/type.pug [hunk 1]
```diff
@@ -13,7 +13,14 @@ mixin typeNameString(type, skipRequired)
         var isArray = true;
         var union = null;
 
-        if (name.slice(-2) === '[]') {
+        if (/ \| /.test(name)) {
+            union = name.split(' | ');
+            if (union) {
```
Comment (connor4312) [inline src/rest/type.pug]: This is always true

File: src/rest/type.pug [hunk 2]
```diff
@@ -42,15 +45,18 @@ mixin typeNameString(type, skipRequired)
...
                 = ' ∪ '
-            if rest.types[nested]
-                a(href=`#${nested}`)= nested
+            -
+                var nestedType = nested;
+                if (nestedType.slice(-2) === '[]') {
+                    nestedType = nested.slice(0,-2);
+                }
+            if rest.types[nestedType]
+                a(href=`#${nestedType}`)= nested
```
Comment (connor4312) [inline src/rest/type.pug]: This is sort of a duplicate of above code. Seems like a simple recursive sort of thing would be useful here.

```python
def print_type(name):
  if is_union(name):
    return ' | '.join(print_type(t) for t in name.split(' | '))
  
  # everything else...
      
```

Comment (ProbablePrime) [inline src/rest/type.pug]: Agreed!

**Comment by ProbablePrime** [comment]
Updated this to use `+printType`

Also swapped interactive-python to interactive-java

File: build/libraries.json [hunk 3]
```diff
@@ -36,9 +36,9 @@
         "official": true
     },
     {
-        "language": "python",
-        "name": "Mixer/interactive-python",
-        "alias": "interactive-python",
+        "language": "java",
+        "name": "Mixer/interactive-java",
+        "alias": "interactive-java",
```
Comment (EthanWaite) [inline build/libraries.json]: We have two called interactive-java now -- could we remove the old one from here?

File: src/rest/type.pug [hunk 4]
```diff
@@ -77,8 +67,18 @@ mixin typeNameString(type, skipRequired)
...
+                +printType(parentType)
         | :
+mixin printType(type)
+    -
+        var typeLink = type;
+        if (typeLink.slice(-2) === '[]') {
+            typeLink = type.slice(0,-2);
+        }
+    if rest.types[typeLink]
+        a(href=`#${typeLink}`)= type
```
Comment (EthanWaite) [inline src/rest/type.pug]: `/rest.html#${typeLink}` please :) the constellation docs use this function.

File: build/libraries.json (hunk 3 above)
Comment (ProbablePrime) [inline build/libraries.json]: Ahh...yes

**Pull request merged**